packages/core/src/browser/shell/side-panel-handler.ts

```diff
@@ -385,12 +393,30 @@ export class SidePanelHandler {
         this.dockPanel.addWidget(widget);
     }
 
+    /**
+     * Add a menu to the sidebar top.
+     *
+     * If the menu is already added, it will be ignored.
+     */
+    addTopMenu(menu: SidebarMenu): void {
+        this.topMenu.addMenu(menu);
+    }
+
+    /**
+     * Remove a menu from the sidebar top.
+     *
+     * @param menuId id of the menu to remove
+     */
+    removeTopMenu(menuId: string): void {
+        this.topMenu.removeMenu(menuId);
+    }
+
     /**
      * Add a menu to the sidebar bottom.
      *
      * If the menu is already added, it will be ignored.
      */
-    addMenu(menu: SidebarBottomMenu): void {
+    addBottomMenu(menu: SidebarMenu): void {
```

Comment:
This is a breaking change for anyone calling `addMenu` and should be documented in the `changelog`.